go/vt/vttablet/grpctmclient/client.go

```diff
@@ -1002,12 +1002,29 @@ func (client *Client) Backup(ctx context.Context, tablet *topodatapb.Tablet, req
 }
 
 // CheckThrottler is part of the tmclient.TabletManagerClient interface.
+// It always tries to use a cached client via the dialer pool as this is
+// called very frequently between tablets when the throttler is enabled in
+// a keyspace and the overhead of creating a new gRPC connection/channel
+// and dialing the other tablet every time is not practical.
 func (client *Client) CheckThrottler(ctx context.Context, tablet *topodatapb.Tablet, req *tabletmanagerdatapb.CheckThrottlerRequest) (*tabletmanagerdatapb.CheckThrottlerResponse, error) {
-	c, closer, err := client.dialer.dial(ctx, tablet)
-	if err != nil {
-		return nil, err
+	var c tabletmanagerservicepb.TabletManagerClient
+	var err error
+	if poolDialer, ok := client.dialer.(poolDialer); ok {
+		c, err = poolDialer.dialPool(ctx, tablet)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I can see the same pattern is used in `ExecuteFetchAsDba` and in `ExecuteFetchAsApp` - so I'm assuming this is correct -- but still unsure -- I'm not sure I understand how this gets a pooled connections;  and I don't see the pattern of "get from pool ; defer return to pool", so this does not read like a way of using a cached connection. Maybe `ExecuteFetchAsApp`'s implementation is likewise incorrect?

Reply:
You can see the implementation within this file: https://github.com/vitessio/vitess/blob/a207a69f5b17d18f305be9eadd3ec6958ee93bf0/go/vt/vttablet/grpctmclient/client.go#L155-L202

Reply:
vs dial that was previously being used which creates the tabletmanagerclient with a new gRPC connection each time: https://github.com/vitessio/vitess/blob/a207a69f5b17d18f305be9eadd3ec6958ee93bf0/go/vt/vttablet/grpctmclient/client.go#L140-L153